diff --git a/CMakeLists.txt b/CMakeLists.txt
index 29b8bfe..8418346 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,8 +34,13 @@ if(REDIS_PLUS_PLUS_BUILD_ASYNC)
         message(STATUS "redis-plus-plus build async interface with libuv")

         # libuv dependency
-        find_path(REDIS_PLUS_PLUS_ASYNC_LIB_HEADER NAMES uv.h)
-        find_library(REDIS_PLUS_PLUS_ASYNC_LIB uv)
+        list(APPEND REDIS_PLUS_PLUS_ASYNC_LIB_HEADER ${CONAN_INCLUDE_DIRS_LIBUV})
+        if(USING_UV_SHARED)
+            list(APPEND REDIS_PLUS_PLUS_ASYNC_LIB uv)
+        else()
+            list(APPEND REDIS_PLUS_PLUS_ASYNC_LIB uv)
+        endif()
+        list(APPEND REDIS_PLUS_PLUS_ASYNC_LIB ${CONAN_LIBS_LIBUV})
     else()
         message(FATAL_ERROR "invalid REDIS_PLUS_PLUS_BUILD_ASYNC")
     endif()
@@ -115,7 +120,6 @@ if(hiredis_FOUND)
     list(APPEND REDIS_PLUS_PLUS_HIREDIS_LIBS hiredis::hiredis)
 
     if(REDIS_PLUS_PLUS_USE_TLS)
-        find_package(hiredis_ssl REQUIRED)
         list(APPEND REDIS_PLUS_PLUS_HIREDIS_LIBS hiredis::hiredis_ssl)
     endif()
 else()
